application/src/main/kotlin/com/studentcenter/weave/application/chat/port/inbound/GetChatMessage.kt

```diff
@@ -0,0 +1,30 @@
+package com.studentcenter.weave.application.chat.port.inbound
+
+import com.studentcenter.weave.domain.chat.entity.ChatMessage
+import com.studentcenter.weave.support.common.dto.ScrollRequest
+import com.studentcenter.weave.support.common.dto.ScrollResponse
+import java.util.UUID
+
+interface GetChatMessage {
+
+    fun getScrollList(query: ScrollListQuery): ScrollListResult
+
+    data class ScrollListQuery(
```

Comment:
p3) 이거 상속 말고 그냥 제너릭만 써서 하면 좀 그럴까요? 
딱히 여기서 문제는 아님

Reply:
상속말고 제너릭만 사용하면 어떤 방식으로 코드가 작성되나요?

Reply:
이런 느낌으로 제너릭만 설정해주면 해당 값만 잘 맞춰서 넣어줄 수 있도록?
```suggestion
interface GetChatMessage {

    fun getScrollList(query: ScrollRequest<UUID>):  ScrollResponse<ChatMessage, UUID?>
}
```

Reply:
음 그렇게 하면 roomId같은 속성은 추가로 파라미터 더 만들어줘야하는데 그게 더 낫다고 생각하시나요?

Reply:
잘 이해를 못 했어요! UUID가 `chatRoomId` 아니에요? 어차피 들어오는 cursor 역할 하는 친구의 타입만 바뀌는 거 같아서요!

Reply:
아뇨 커서역할은 messageId이구요, chatRoomId는 where 조건절중 하나에요

Reply:
아하 그러면 그냥 DataClass 싸서 제너릭으로 넘기기?

Reply:
scrolling에 대한 처리는 그냥 ScrollRequest에서 하면 될 것 같고 필요한 데이터만 정의해서 넘겨주는 정도?
필요하면 interface로 제너릭을 Cursor로 만들고 getCursor() 같은 것들 만들어도 되고?

Reply:
data class싸서 제너릭으로 넘긴다는말이 무슨뜻인지 잘모르겠어요, 
next나 cursor나 동일한 역할이라 이름은 굳이 안바꿔도 될것같은데

Reply:
next의 타입을 데이터 클래스로 두고 쓰자는 말씀이실까요? 채팅방 아이디를 사용해야 한다면 next.chatRoomId 이런식으로 사용해야해서 개인적으로 이방법은 좀 직관적이지는 않은느낌?

Reply:
제가 제안하고 싶은 건 지금은 거의 사실 Scroll관련 요청에서 next조건이 되는 타입 필드만 정의해서 쓰는 정도라 굳이 상속 안하고 제너릭으로 하고 클래스에 잘 미리 정의해서 재활용하는게 낫지 않나에요!

Reply:
밑에 `ScrollListResult`도 그냥 ScrollResponse<CHatMessage, UUID?> 를 그대로 써도 될 것 같아서요!

Reply:
어 근데 다른 ScrollResponse, ScrollRequest 활용케이스들은 모두 상속해서 정의하고 있는데, 이부분에 대해서만 예외적으로 적용하는게 일관성이 좀 깨질것같은데, 어떻게 생각하시나요?

Reply:
네네 맞아요 그래서 괜찮으면 한번에 리팩토링 할까 생각했어요!

Reply:
아하 요거는 나중에 조금더 논의하면 좋을거같아요